mixer/pkg/api/envoyGrpcServer.go

```diff
@@ -0,0 +1,243 @@
+		globalWordList []string
+		globalDict     map[string]int32
+
+		// load shedding
+		throttler *loadshedding.Throttler
+	}
+)
+
+func NewGRPCServerEnvoy(dispatcher dispatcher.Dispatcher, gp *pool.GoroutinePool, cache *checkcache.Cache,
+	throttler *loadshedding.Throttler) GrpcServerEnvoy {
+	list := attribute.GlobalList()
+	globalDict := make(map[string]int32, len(list))
+	for i := 0; i < len(list); i++ {
+		globalDict[list[i]] = int32(i)
+	}
+	return GrpcServerEnvoy{
+		dispatcher:     dispatcher,
+		gp:             gp,
+		globalWordList: list,
+		globalDict:     globalDict,
+		cache:          cache,
+		throttler:      throttler,
+	}
+}
+
+// Mirrors Mixer Check but instead uses Envoy External Authorization API
+// It enables longevity of OOP adapters
+//https://www.envoyproxy.io/docs/envoy/latest/api-v2/config/filter/http/ext_authz/v2/ext_authz.proto
+func (s *GrpcServerEnvoy) Check(ctx context.Context, req *authzGRPC.CheckRequest) (*authzGRPC.CheckResponse, error) {
+
+	if s.throttler.Throttle(loadshedding.RequestInfo{PredictedCost: 1.0}) {
+		return nil, grpc.Errorf(codes.Unavailable, "Envoy Server is currently overloaded. Please try again.")
+	}
+
+	envoyProtoBag := attribute.AuthzProtoBag(req)
+	if s.cache != nil {
+		if value, ok := s.cache.Get(envoyProtoBag); ok {
+			var resp *authzGRPC.CheckResponse
+			cacheStatus := rpc.Status{
+				Code:    value.StatusCode,
+				Message: value.StatusMessage,
+			}
+			if status.IsOK(cacheStatus) {
+				lg.Debug("ExtAuthz.Check approved")
+				resp = &authzGRPC.CheckResponse{
+					Status: &rpcstatus.Status{
+						Code:    value.StatusCode,
+						Message: value.StatusMessage,
+					},
+					HttpResponse: &authzGRPC.CheckResponse_OkResponse{
+						OkResponse: &authzGRPC.OkHttpResponse{},
+					},
+				}
+			} else {
+				lg.Debugf("ExtAuthz.Check denied: %v", value.StatusCode)
+				resp = &authzGRPC.CheckResponse{
+					Status: &rpcstatus.Status{
+						Code:    value.StatusCode,
+						Message: value.StatusMessage,
+					},
+					HttpResponse: &authzGRPC.CheckResponse_DeniedResponse{
+						DeniedResponse: &authzGRPC.DeniedHttpResponse{},
+					},
+				}
+
+			}
+
+			lg.Debugf("ExtAuthz.Check() status from cache: %v", resp.Status)
+
+			if !status.IsOK(cacheStatus) {
+				return resp, nil
+			}
+		}
+	}
+	envoyCheckBag := attr.GetMutableBag(envoyProtoBag)
+	resp, err := s.checkEnvoy(ctx, envoyProtoBag, envoyCheckBag)
+
+	envoyProtoBag.Done()
+	envoyCheckBag.Done()
+	return resp, err
+}
+
+func (s *GrpcServerEnvoy) checkEnvoy(ctx context.Context, protoBag attr.Bag, checkBag *attr.MutableBag) (*authzGRPC.CheckResponse, error) {
+
+	if err := s.dispatcher.Preprocess(ctx, protoBag, checkBag); err != nil {
+		err = fmt.Errorf("preprocessing attributes failed: %v", err)
+		lg.Errora("ExtAuthz.Check failed: ", err.Error())
+		return nil, grpc.Errorf(codes.Internal, err.Error())
+	}
+
+	lg.Debug("Dispatching to main adapters after running processors")
```

Comment:
nit: i'd just remove this line. i've never liked the "main adapters" bit and subsequent log lines cover the dispatch bit.